Pin third party GitHub actions (#3027)
hyperspy/hyperspy: doc/dev_guide/maintenance.rst

.. _maintenance-label:

Maintenance
===========

> GitHub Workflow
> ^^^^^^^^^^^^^^^

Reviewer: ```suggestion
GitHub Workflows
^^^^^^^^^^^^^^^^
```

> `GitHub workflow <https://github.com/hyperspy/hyperspy/actions>`_ are used to:

Reviewer: ```suggestion
`GitHub workflows <https://github.com/hyperspy/hyperspy/actions>`_ are used to:
```

 * run the test suite
 * build packages and upload to pypi and GitHub release
 * build the documentation and check the links

Some of these workflow need to access `GitHub "secrets" <https://docs.github.com/en/actions/security-guides/encrypted-secrets>`_,
which are private to the HyperSpy repository, in order to be able to upload to pypi or the
`GITHUB_TOKEN <https://docs.github.com/en/actions/security-guides/automatic-token-authentication>`_
to push code to the other branches.
To reduce the risk that these "secrets" are made accessible publicly, for example, through the

> injection of malicious code by third party in one of the GitHub workflow used in the HyperSpy

Reviewer: ```suggestion
injection of malicious code by third parties in one of the GitHub workflows used in the HyperSpy
```

organisation, the third party actions (those that are not provided by established trusted parties)
are pinned to the ``SHA`` of a specific commit, which is trusted not to contain malicious code.

Updating GitHub Actions
^^^^^^^^^^^^^^^^^^^^^^^

The workflows in the HyperSpy repository use GitHub actions provided by established trusted parties
and third parties. They are updated regularly by the
`dependabot <https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuring-dependabot-version-updates>`_
in pull requests.

> When updating a third party action, the action have to be pinned using the ``SHA`` of the commit of

Reviewer: ```suggestion
When updating a third party action, the action has to be pinned using the ``SHA`` of the commit of
```
